### ISSUE TITLE
Please use external package for 'catch'

### ISSUE THREAD

**Issue by yurivict** [rdkit/rdkit#2133]
```External/catch/CMakeLists.txt``` currently downloads it.
Downloading files during build isn't allowed in most packaging systems (ex. FreeBSD, Redhat).
The reason for this is security: direct downloads are generally unsafe.

At the same time, ```catch``` is available as a separate package in almost all systems: https://repology.org/metapackage/catch/versions

Please change to the external package.

**Comment by greglandrum** [comment]
I agree that there should be a way to build the code and use an existing catch installation (same with the coordgen and maeparser integration). This is something we should try and get cleaned up during this release cycle.

**Comment by bp-kelley** [comment]
For coordgen and maeparser, could we use git submodules?

**Comment by yurivict** [comment]
FreeBSD has packages for coordgen and maeparser.
Can you use sumbodules or use external packages based on the corresponding cmake options?

**Comment by e-kwsm** [comment]
System's Catch2 is already available: #2327.
See #2780 for `git submodule`.

**Comment by yurivict** [comment]
not stale
